File: 1_Two_Sum/1_Two_Sum.py
__author__ = "CodeamonCat"
__email__ = "CodeamonCat@gmail.com"


class Solution:

    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_dict = dict()  # key: nums, value: nums_index
        for index, value in enumerate(nums):
            complement = target - value
            if nums_dict.__contains__(complement):
                return [nums_dict[complement], index]
            else:
                nums_dict[value] = index
        return [-1, -1]

File: 1_Two_Sum/1_Two_Sum.java
/*
 * @author CodeamonCat/CodeamonCat@gmail.com
 */

class Solution {
  public int[] twoSum(int[] nums, int target) {
    HashMap<Integer, Integer> hashtable = new HashMap<>();
    for (int index = 0; index < nums.length; index++) {
      int complement = target - nums[index];
      if (hashtable.containsKey(complement)) {
        return new int[] {hashtable.get(complement), index};
      }
      hashtable.put(nums[index], index);
    }
    return new int[] {};
  }
}
